stablehlo/dialect/StablehloBytecode.cpp

```diff
@@ -491,30 +491,26 @@ GatherDimensionNumbersAttr
 StablehloBytecodeInterface::readGatherDimensionNumbersAttr(
     DialectBytecodeReader &reader) const {
   LOG_READ_CALL;
-  llvm::SmallVector<int64_t> offsetDims, collapsedSliceDims,
-      operandBatchingDims, startIndicesBatchingDims, startIndexMap;
+  llvm::SmallVector<int64_t> offsetDims, collapsedSliceDims, startIndexMap;
   int64_t indexVectorDim;
 
   if (failed(reader.readSignedVarInts(offsetDims)) ||
       failed(reader.readSignedVarInts(collapsedSliceDims)) ||
-      failed(reader.readSignedVarInts(operandBatchingDims)) ||
-      failed(reader.readSignedVarInts(startIndicesBatchingDims)) ||
       failed(reader.readSignedVarInts(startIndexMap)) ||
       failed(reader.readSignedVarInt(indexVectorDim)))
     return GatherDimensionNumbersAttr();
 
   return GatherDimensionNumbersAttr::get(
-      getContext(), offsetDims, collapsedSliceDims, operandBatchingDims,
-      startIndicesBatchingDims, startIndexMap, indexVectorDim);
+      getContext(), offsetDims, collapsedSliceDims,
+      /*operandBatchingDims=*/{}, /*startIndicesBatchingDims=*/{},
+      startIndexMap, indexVectorDim);
 }
 
 void StablehloBytecodeInterface::write(GatherDimensionNumbersAttr attr,
                                        DialectBytecodeWriter &writer) const {
   writer.writeVarInt(stablehlo_encoding::kGatherDimensionNumbers);
   writer.writeSignedVarInts(attr.getOffsetDims());
   writer.writeSignedVarInts(attr.getCollapsedSliceDims());
-  writer.writeSignedVarInts(attr.getOperandBatchingDims());
-  writer.writeSignedVarInts(attr.getStartIndicesBatchingDims());
```

Comment:
Should we report fatal if these aren't set? Currently they're silently dropped I believe

Reply:
Good point! I need to get these changes in, but I will look into that and send a follow-up.